client/packages/common/src/ui/layout/tables/components/Cells/StatusCell/StatusCell.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import { CellProps } from '../../../columns';
+import { Box, CircleIcon, Typography } from '@openmsupply-client/common';
+import { RecordWithId } from '@common/types';
+
+interface StatusProps {
+  statusMap: Record<
+    string,
+    { label: string; color: string; bgColor?: string }
+  > | null;
+}
+
+export const StatusCell = <T extends RecordWithId>({
+  rowData,
+  column,
+  statusMap,
+}: CellProps<T> & StatusProps) => {
+  const { label, color, bgColor } =
+    statusMap?.[String(column.accessor({ rowData }))] ?? {};
+
+  if (!label) return null;
+  return (
+    <Box
+      sx={{
+        borderRadius: 4,
+        textAlign: 'center',
+        backgroundColor: bgColor,
+      }}
+      paddingY={0.1}
+      paddingX={0.5}
+      display="flex"
+      alignItems="center"
+      width="fit-content"
+      minWidth={100}
+      position="relative"
+    >
+      {!bgColor && (
+        <Box
+          sx={{
+            backgroundColor: color,
+            width: '100%',
+            height: '100%',
+            position: 'absolute',
+            left: 0,
+            top: 0,
+            borderRadius: 4,
+            opacity: 0.2,
+          }}
+        />
+      )}
+      <CircleIcon
+        sx={{
+          color,
+          transform: 'scale(0.4)',
+        }}
+      />
+      <Typography sx={{ paddingRight: 1 }}>{label}</Typography>
+    </Box>
```

Comment:
This is a combination of the Cold chain "Status" cell (the "chip") and the "DotCell" (the dot icon). You have the option to specify a background colour explicitly, but by default it will use a faded-out version of the main colour. I did this so we could just use existing palette colours and not have to define a bunch of new background colours. But perhaps made the logic a little over-complicated: when bgColor is defined, we can apply it to the main container, but if not, we have to apply it to a separate, absolutely positioned block, otherwise the entire component gets the low-opacity effect.

Reply:
It might be simpler to always render/user the separate Box, with styles something like

```ts
          backgroundColor: bgColor ?? color,
          opacity: bgColor ? 1 : 0.2,
```

rather than setting the background colour in two different places?

Reply:
Yup, good idea: 7ed8975781